design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,527 @@
+    firstname: String @doc(description: "Sales representative first name.")
+    lastname: String @doc(description: "Sales representative last name.")
+}
+
+type CompanyUsers @doc(description: "Output data schema for an object returned by a Company users search query.") {
+    items: [CompanyUser] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "An object that includes the pagination meta data of the search query.")
+}
+
+type CompanyUser @doc(description: "Company User (Customer assigned to Company) entity output data schema.") {
+    id: Int @doc(description: "Company User/Customer ID.")
+    email: String @doc(description: "Company User email address.")
+    firstname: String @doc(description: "Company User first name.")
+    lastname: String @doc(description: "Company User last name.")
+    status: Int @doc(description: "Company User status ID.")
+    status_name: String @doc(description: "Company User status name.")
+    job_title: String @doc(description: "Company User job title.")
+    telephone: String @doc(description: "Company User phone number.")
+    role: CompanyRole @doc(description: "Company User role data (includes permissions).")
+    team: CompanyTeam @doc(description: "Company User team data.")
+}
+
+type CompanyRoles @doc(description: "Output data schema for an object returned by a Company roles search query.") {
+    items: [CompanyRole] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "An object that includes the pagination meta data of the search query.")
+}
+
+type CompanyRole @doc(description: "Company role output data schema returned in response to a query by Role ID.") {
+    id: Int @doc(description: "Role ID.")
+    name: String @doc(description: "Role name.")
+    users_count: Int @doc(description: "Total number of Users with such Role within Company Hierarchy.")
+    permissions: [String] @doc(description: "A list of permission resources defined for a Role.")
+}
+
+type CompanyAclResource @doc(description: "Output data schema for an object with Role permission resource information.") {
+    id: String @doc(description: "ACL resource ID.")
+    text: String @doc(description: "ACL resource label.")
+    sortOrder: Int @doc(description: "ACL resource sort order.")
+    children: [CompanyAclResource!] @doc(description: "An array of sub-resources.")
+}
+
+type CompanyUserEmailCheckResponse @doc(description: "Response object schema for a Company User email validation query.") {
+    status: Int @doc(description: "1 - 'OK', 0 - 'Error'")
+    message: String @doc(description: "Error or notice message for a User.")
```

Comment:
This seem to be excessive, we can rely on built-in error reporting functionality instead of introducing custom fields for errors.
The only case when we need custom error fields is when we have multiple fields validated at once and need to report all validation errors.

Reply:
fixed